src/openrct2/ride/Track.cpp

```diff
@@ -1491,9 +1491,10 @@ static money32 track_place(sint32 rideIndex,
         tileElement->type = map_type;
 
         tile_element_set_track_sequence(tileElement, trackBlock->index);
-        tileElement->properties.track.ride_index = rideIndex;
-        tileElement->properties.track.type       = type;
-        tileElement->properties.track.colour     = 0;
+        track_element_set_ride_index(tileElement, rideIndex);
+        track_element_set_type(tileElement, type);
+        // TODO: Find out why this is set to 0.
```

Comment:
`track.colour` is not just for colour. It is also used for a bunch of flags. This zeros those flags `TRACK_ELEMENT_COLOUR_FLAG_...`.

Reply:
I know what it is used for. After all, I abstracted almost all access to it.
What I'm unsure of is if I need to set all of those properties, or if only some of them would be enough,

Reply:
Tile elements are not cleared when they are deleted so i would say it needs to be reset to zero all of the flags. The colour gets set later so no need to zero that.

Reply:
Their properties are memset to 0 when creating them with tile_element_insert(). So this appears to be redundant.